config-example.yaml

```diff
@@ -40,6 +40,13 @@ grpc_listen_addr: 127.0.0.1:50443
 # are doing.
 grpc_allow_insecure: false
 
+# The Access-Control-Allow-Origin header specifies which origins are allowed to access resources.
+# Options:
+# - "*" to allow access from any origin (not recommended for sensitive data).
+# - "http://example.com" to only allow access from a specific origin.
+# - "" to disable Cross-Origin Resource Sharing (CORS).
+Access-Control-Allow-Origin: ""
```

Comment:
This does not follow the format of the config file and has not been discussed. This isnt really anything we need and people needing this can handle it in Nginx and the likes.

Reply:
Hi Kradalby,

I have modified the code to align with the requested configuration file format. 

I did open an accompanying issue for discussion: #2301. I also see mentions of a project chat, could you point me to it?

The idea of this feature is to simplify the Headscale server setup and have everything in one place. Tailscale continues to grow in the browser sector; for example, they recently released their new SSH client. Our company has just released WebVM 2.0, which is a virtual machine inside the browser. We use this product with Tailscale to establish fast peer-to-peer connections between machines. With this small change, the Headscale setup for the browser would be much easier and much simpler, especially in combination with your integrated TLS support. This feature has already been requested in two other issues but without success. Here are their references: #623 #1283 .

Reply:
> This does not follow the format of the config file and has not been discussed. This isnt really anything we need and people needing this can handle it in Nginx and the likes.

FYI expecting people to rely on a reverse proxy to intercept requests and inject CORS is not normally acceptable behavior. I (as the maintainer of headscale ui) get people confused about this pretty much twice a month and opening issues as a result. It also results in people who aren't just relying on the browser to ignore CORS entirely (IE: when making calls via node instead), which lowers security within both applications.

A simple `UI_DOMAIN` configuration that optionally sets the appropriate CORS information would solve a lot of issues. I would not suggest allowing `*` anyway since browsers are planning to block that syntax.

Reply:
> A simple `UI_DOMAIN` configuration that optionally sets the appropriate CORS information would solve a lot of issues. I would not suggest allowing `*` anyway since browsers are planning to block that syntax.

We do want to keep the flexibility of using the `*` available, but if indeed browsers are planning to block this syntax, I will make sure to update the feature to check for this syntax. To avoid any confusion, I would prefer to stick with the normal CORS syntax. This `*` is indeed another security risk, but this flexibility is also possible by using a reverse proxy. However, if Headscale agrees to not allow `*`, then I would gladly make another commit with a fix.